pgmax/fg/graph.py

```diff
@@ -1102,15 +1152,18 @@ def decode_map_states(beliefs: Any) -> Any:
     Returns:
         An array or a PyTree container containing the MAP states for different variables.
     """
-    map_states = jax.tree_util.tree_map(
-        lambda x: jnp.argmax(x, axis=-1),
-        beliefs,
-    )
+
+    @jax.jit
+    def _decode_map_states(beliefs) -> Any:
+        return jax.tree_util.tree_map(lambda x: jnp.argmax(x, axis=-1), beliefs)
+
+    map_states = {}
+    for variable_group, vgroup_beliefs in beliefs.items():
+        map_states[variable_group] = _decode_map_states(vgroup_beliefs)
     return map_states
```

Comment:
Why is this necessary? `beliefs` should be a pytree and `tree_map` should be directly applicable. If that's not the case we should figure out why instead of making this hack